core/src/com/unciv/ui/pickerscreens/TechButton.kt

```diff
@@ -40,34 +40,36 @@ class TechButton(techName:String, val techManager: TechManager, isWorldScreen: B
     private fun addTechEnabledIcons(techName: String, isWorldScreen: Boolean, rightSide: Table) {
         val techEnabledIcons = Table()
         techEnabledIcons.defaults().pad(5f)
+        val techIconSize = 30f
 
         val civName = techManager.civInfo.civName
         val gameBasics = techManager.civInfo.gameInfo.ruleSet
 
         val tech = gameBasics.technologies[techName]!!
 
         for (unit in tech.getEnabledUnits(techManager.civInfo))
-            techEnabledIcons.add(ImageGetter.getConstructionImage(unit.name).surroundWithCircle(30f))
+            techEnabledIcons.add(ImageGetter.getConstructionImage(unit.name).surroundWithCircle(techIconSize))
 
         for (building in tech.getEnabledBuildings(techManager.civInfo))
-            techEnabledIcons.add(ImageGetter.getConstructionImage(building.name).surroundWithCircle(30f))
+            techEnabledIcons.add(ImageGetter.getConstructionImage(building.name).surroundWithCircle(techIconSize))
 
         for (improvement in gameBasics.tileImprovements.values
                 .filter { it.techRequired == techName || it.improvingTech == techName }
                 .filter { it.uniqueTo==null || it.uniqueTo==civName }) {
             if (improvement.name.startsWith("Remove"))
-                techEnabledIcons.add(ImageGetter.getImage("OtherIcons/Stop")).size(30f)
-            else techEnabledIcons.add(ImageGetter.getImprovementIcon(improvement.name, 30f))
+                techEnabledIcons.add(ImageGetter.getImage("OtherIcons/Stop")).size(techIconSize)
+            else techEnabledIcons.add(ImageGetter.getImprovementIcon(improvement.name, techIconSize))
         }
 
         for (resource in gameBasics.tileResources.values.filter { it.revealedBy == techName })
-            techEnabledIcons.add(ImageGetter.getResourceImage(resource.name, 30f))
+            techEnabledIcons.add(ImageGetter.getResourceImage(resource.name, techIconSize))
 
         for (unique in tech.uniques)
             techEnabledIcons.add(ImageGetter.getImage("OtherIcons/Star")
-                    .apply { color = Color.BLACK }.surroundWithCircle(30f))
+                    .apply { color = Color.BLACK }.surroundWithCircle(techIconSize))
 
         if (isWorldScreen) rightSide.add(techEnabledIcons)
-        else rightSide.add(techEnabledIcons).width(150f)
+        else rightSide.add(techEnabledIcons).width(
```

Comment:
I think we can just do minWidth instead, and then the table will automatically adjust to the techEnabledIcons when relevant

Reply:
That sounds as a good idea, however, I did not find a good technical way to do that.
`Table` (i.e. `rightSide` and `techEnabledIcons`) does not have a setter for `minWidth`.

Reply:
I try to do that for the given cell, and it doesn't look much differ. (see the commit)